pinot-core/src/main/java/com/linkedin/pinot/core/segment/index/ColumnMetadata.java

```diff
@@ -328,17 +397,16 @@ public char getPaddingCharacter() {
     return paddingCharacter;
   }
 
```

Comment:
We have getDerivedMetricType(), from which outside methods can get the type info. And if in the future, there exists other derived types, outside methods can add a new "case" clause in switch for handling that type, which seems clear.